virttest/qemu_devices/qcontainer.py

```diff
@@ -261,19 +267,33 @@ def allocate_iothread(self, iothread, device):
 
     def _probe_capabilities(self):
         """ Probe capabilities. """
-        ver = utils_qemu.get_qemu_version(self.__qemu_binary)[0]
-
         # -blockdev
         if (self.has_option('blockdev') and
-                ver in VersionInterval(self.BLOCKDEV_VERSION_SCOPE)):
+                self.__qemu_ver in VersionInterval(self.BLOCKDEV_VERSION_SCOPE)):
             self.caps.set_flag(Flags.BLOCKDEV)
         # -smp dies=?
-        if ver in VersionInterval(self.SMP_DIES_VERSION_SCOPE):
+        if self.__qemu_ver in VersionInterval(self.SMP_DIES_VERSION_SCOPE):
             self.caps.set_flag(Flags.SMP_DIES)
         # -incoming defer
         if self.has_option('incoming defer'):
             self.caps.set_flag(Flags.INCOMING_DEFER)
 
+        if (self.has_qmp_cmd('migrate-set-parameters') and
+                self.has_hmp_cmd('migrate_set_parameter')):
+            self.caps.set_flag(Flags.MIGRATION_PARAMS)
+
+    def _probe_migration_parameters(self):
+        """Probe migration parameters."""
+        mapping = {
+            MigrationParams.DOWNTIME_LIMIT: self.MIGRATION_DOWNTIME_LIMTT_VERSION_SCOPE,
+            MigrationParams.MAX_BANDWIDTH: self.MIGRATION_MAX_BANDWIDTH_VERSION_SCOPE,
+            MigrationParams.XBZRLE_CACHE_SIZE: self.MIGRATION_XBZRLE_CACHE_SIZE_VERSION_SCOPE,
+        }
+
+        for mig_param, ver_scope in mapping.items():
+            if self.__qemu_ver in VersionInterval(ver_scope):
+                self.caps.set_flag(mig_param)
```

Comment:
Let's introduce a new property to store these caps. e.g. `self.mig_params`

Reply:
Thanks, have updated.